common/src/main/java/com/google/udmi/util/JsonUtil.java

```diff
@@ -486,4 +489,121 @@ public static void writeFile(Object theThing, File file) {
       throw new RuntimeException("While writing " + file.getAbsolutePath(), e);
     }
   }
+
+  public static Map<String, Object> flattenNestedMap(Map<String, Object> map, String separator) {
+    Map<String, Object> flattenedMap = new TreeMap<>();
+    flatten(map, "", flattenedMap, separator);
+    return flattenedMap;
+  }
+
+  @SuppressWarnings("unchecked")
+  private static void flatten(Map<String, Object> currentMap, String currentKey,
+      Map<String, Object> flattenedMap, String separator) {
+    for (Map.Entry<String, Object> entry : currentMap.entrySet()) {
+      String key = entry.getKey();
+      Object value = entry.getValue();
+      String newKey = currentKey.isEmpty() ? key : currentKey + separator + key;
+
+      if (value instanceof Map) {
+        flatten((Map<String, Object>) value, newKey, flattenedMap, separator);
+      } else {
+        flattenedMap.put(newKey, value);
+      }
+    }
+  }
+
+  public static JsonNode nestFlattenedJson(Map<String, String> flattenedJsonMap) {
+    ObjectMapper mapper = new ObjectMapper();
```

Comment:
There are pre-existing constants for various ObjectMappers at the top of this file... so likely best to use one of those.